lib/checks/keyboard/main-is-top-level.js

```diff
@@ -0,0 +1,13 @@
+var landmarks = axe.commons.aria.getRolesByType('landmark');
+var parent = node.parentElement;
+while (parent){
+	var role = parent.getAttribute('role');
+	if (!role){
+		role = axe.commons.aria.implicitRole(parent);
```

Comment:
I think this is too strict. There are roles that can certainly be permitted, off the top of my head: `form`, `application`, `presentation`. Probably a bunch more. Can you go through the list and make sure those kind of things get ignored?